sklearn/linear_model/coordinate_descent.py

```diff
@@ -696,28 +727,64 @@ def fit(self, X, y, check_input=True):
                           "well. You are advised to use the LinearRegression "
                           "estimator", stacklevel=2)
 
+        if (not isinstance(self.l1_ratio, numbers.Number) or
+                self.l1_ratio < 0 or self.l1_ratio > 1):
+                    raise ValueError("l1_ratio must be between 0 and 1;"
+                                     " got (l1_ratio=%r)" % self.l1_ratio)
+
+        all_solvers = ['cd', 'saga']
+        if self.solver not in all_solvers:
+            raise ValueError("ElasticNet Regression supports only solvers in"
+                             " {}, got {}.".format(all_solvers, self.solver))
+
         if isinstance(self.precompute, str):
             raise ValueError('precompute should be one of True, False or'
                              ' array-like. Got %r' % self.precompute)
+        elif self.precompute is not False and self.solver == 'saga':
```

Comment:
I don't think it's wrong, but `self.precomute != False` (or `not self.precompute` if `None` should be considered the same here) seems more natural to me.

Reply:
`self.precompute != False` gives "E712 comparison to False should be 'if cond is not false' or ...".
I personally find `if not self.precompute` less readable than the `if self.precompute is not False`.